### PR TITLE
Add JSON representation of index and group overview

### PR DESCRIPTION
**Pull request by Martchus**
JSON response can be triggered by adding HTTP header 'Accept: application/json'

Eg.:
```
curl -H 'Accept: application/json' http://localhost:9526?time_limit_days=48 | json_pp
curl -H 'Accept: application/json' http://localhost:9526/group_overview/25?time_limit_days=48 | json_pp
```

Like the former approach, this is meant for the openQA review script.

I don't know how to add the magic for `.json` routes but this variant (proposed by @coolo) looks good as well in my opinion.

Note that this can not be done generically for each page because currently database objects are stashed and those are not rendered as JSON automatically.

### CODE REVIEW

**Comment by Martchus** [comment]
Actually this approach is from @coolo. I just adapted it.

I did not duplicate things, but still some additional work was required (including adjustments in the view code for the HTML rendering).

I'll comment the relevant lines which allow the distinction of the output format.

Also, I think we already cover most of it with the existing tests.

File: lib/OpenQA/WebAPI/Controller/Main.pm [hunk 1]
```diff
@@ -39,6 +39,9 @@ sub index {
         push(@results, $build_results) if $build_results;
     }
     $self->stash('results', \@results);
+    $self->respond_to(
+        json => {json     => {results => \@results}},
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: This line does the magic. As far as I see it this is the only line the current tests don't cover (beside the corresponding line for the group overview).

File: lib/OpenQA/WebAPI/Controller/Main.pm [hunk 2]
```diff
@@ -89,11 +92,13 @@ sub group_overview {
         }
     }
     $self->stash('result',          $res);
-    $self->stash('group',           $group);
     $self->stash('limit_builds',    $limit_builds);
     $self->stash('only_tagged',     $only_tagged);
     $self->stash('comments',        \@comments);
     $self->stash('pinned_comments', \@pinned_comments);
+    $self->respond_to(
+        json => {json     => {result => $res}},
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: I guess you should put the comments in there too.

Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: I already had the same idea but then I thought this is beyond of the problem this feature tries to solve. We have comments API anyways.

But if you like I include it. I would have to resolve the DB queries in the controller and adjust the view code.

**Comment by coolo** [comment]
See https://github.com/os-autoinst/openQA/pull/925 for a followup

**Pull request merged**